### PR TITLE
Support allowedAppends()

### PR DESCRIPTION
**Pull request by robsontenorio**
# Append attributes

Sometimes you will want to append some custom attributes into result from a Model. This can be done using the `append` parameter.

```
class User extends Model{

    getFullnameAttribute()
    {
        return $this->firstname.' '.$this->lastname;
    }
}
```

```
// GET /users?append=fullname

$users = QueryBuilder::for(User::class)
    ->allowedAppends('fullname')
    ->get();
```

Of course you can pass a list of attributes to be appended.

```
// GET /users?append=fullname,ranking
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -308,6 +308,35 @@ GET /posts?include=author&fields[author]=name
...
 All posts will be fetched including only the name of the author. 
 
+### Append attributes
+
+Sometimes you will want to append some custom attributes into result from a Model. This can be done using the `append` parameter.
+
+```
+class User extends Model{
+
+    getFullnameAttribute()
```
Comment (freekmurze) [inline README.md]: Add the `function` keyword here

File: src/Exceptions/InvalidAppendQuery.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class InvalidAppendQuery extends InvalidQuery
+{
+    /** @var \Illuminate\Support\Collection */
+    public $unknownAppends;
+
+    /** @var \Illuminate\Support\Collection */
+    public $allowedAppends;
+
+    public function __construct(Collection $unknownAppends, Collection $allowedAppends)
+    {
```
Comment (freekmurze) [inline src/Exceptions/InvalidAppendQuery.php]: Let's do all this in the `appendsNotAllowed` function.

**Comment by robsontenorio** [comment]
@freekmurze Done!

**Comment by freekmurze** [comment]
Thank you!

**Pull request merged**